app/views/layouts/darkswarm.html.haml

```diff
@@ -5,6 +5,8 @@
     %meta{property: "og:title", content: content_for?(:title) ? yield(:title) : t(:title)}
     %meta{property: "og:description", content: content_for?(:description) ? yield(:description) : t(:site_meta_description)}
     %meta{property: "og:image", content: content_for?(:image) ? yield(:image) : ContentConfig.logo.url}
+    - if !Rails.env.production? || @noindex_meta_tag
```

Comment:
I'd rather have a method that assigns a hash like `{ name: 'robots', content: 'noindex' }` depending on this boolean logic above and assigns it in an ivar. Then the view is as simple (and decoupled) as:

```ruby
%meta{name: @meta_tag.name, content: @meta_tag.content} if @meta_tag
```

What do you think?

Reply:
I'm not sure that it wouldn't be premature refactoring. Sure it's more decoupled this way, but we also lose some explicitness.

Reply:
Could be. The think I dislike the most is start coupling views with environments though.

Reply:
Ok, so for now can we just move the environment check into a guard clause in the `set_noindex_meta_tag` method? Then we can just have:



````ruby
# controller
def set_noindex_meta_tag
  return if Rails.env.production? && current_distributor.visible?
  @noindex_meta_tag = true
end

...

#view
%meta{name: "robots", content: "noindex"} if @noindex_meta_tag 
````

Reply:
Yes but it changes the condition. From `(not production env OR (shop action
AND invisible shop))` it becomes `(shop action AND (not production env AND
invisible shop))`. What I intended to do was un-index the whole site if it's
not in production.

Reply:
Oh, right! We could add the setter to the base controller and put the env check there, then call super from the EnterprisesController? Getting messy... Maybe it’s ok the way it is?

Reply:
I really think it's the best way because we clearly understand when the tag is "displayed"